node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -1,29 +1,51 @@
 package coop.rchain.node
 
+import java.io.FileWriter
+import java.math.BigInteger
+import java.nio.file.Path
+import java.security.spec.{ECGenParameterSpec, PKCS8EncodedKeySpec}
+import java.security.{KeyFactory, KeyPairGenerator, SecureRandom}
+
+import cats.effect.Resource
+
 import scala.collection.JavaConverters._
 import scala.tools.jline.console._
 import completer.StringsCompleter
-
 import cats.implicits._
-
 import coop.rchain.casper.util.comm._
 import coop.rchain.casper.util.BondingUtil
 import coop.rchain.catscontrib.TaskContrib._
+import coop.rchain.crypto.PrivateKey
 import coop.rchain.crypto.codec.Base16
-import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1, SignaturesAlg}
+import coop.rchain.crypto.util.SecureRandomUtil
 import coop.rchain.metrics
 import coop.rchain.metrics.Metrics
 import coop.rchain.node.configuration._
 import coop.rchain.node.effects._
 import coop.rchain.shared._
 import coop.rchain.shared.StringOps._
-
 import monix.eval.Task
 import monix.execution.Scheduler
+import org.bouncycastle.asn1.{ASN1Sequence, DEROctetString}
+import org.bouncycastle.asn1.edec.EdECObjectIdentifiers
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo
+import org.bouncycastle.asn1.sec.ECPrivateKey
+import org.bouncycastle.asn1.x509.AlgorithmIdentifier
+import org.bouncycastle.asn1.x9.X9ObjectIdentifiers
+import org.bouncycastle.crypto.ec.CustomNamedCurves
+import org.bouncycastle.crypto.util.{PrivateKeyFactory, PrivateKeyInfoFactory}
+import org.bouncycastle.jcajce.provider.asymmetric.util.ECUtil
+import org.bouncycastle.jce.ECNamedCurveTable
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.jce.spec.{ECParameterSpec, ECPrivateKeySpec}
+import org.bouncycastle.openssl.jcajce.{JcaMiscPEMGenerator, JcaPEMWriter, JcePEMEncryptorBuilder}
+import org.bouncycastle.util.io.pem.PemObject
```

Comment:
@itegulov All these new imports show that `Main` does things out of the scope of launching the application. Please move all the encryption related logic into the `crypto` sub-project.

Reply:
Good point